src/game/Server/WorldSocket.cpp

```diff
@@ -766,33 +752,30 @@ int WorldSocket::HandleAuthSession(WorldPacket& recvPacket)
     // Must be done before WorldSession is created
     if (wardenActive && os != "Win" && os != "OSX")
     {
-        WorldPacket Packet(SMSG_AUTH_RESPONSE, 1);
-        Packet << uint8(AUTH_REJECT);
-
+        WorldPacket packet(SMSG_AUTH_RESPONSE, 1);
+        packet << uint8(AUTH_REJECT);
         SendPacket(packet);
 
         BASIC_LOG("WorldSocket::HandleAuthSession: Client %s attempted to log in using invalid client OS (%s).", GetRemoteAddress().c_str(), os.c_str());
         return -1;
     }
 
     // Check that Key and account name are the same on client and server
-    Sha1Hash sha;
-
-    uint32 t = 0;
+    uint8 t[4]{ 0 };
     uint32 seed = m_Seed;
 
+    Sha1Hash sha;
     sha.UpdateData(account);
     sha.UpdateData((uint8*) & t, 4);
     sha.UpdateData((uint8*) & clientSeed, 4);
     sha.UpdateData((uint8*) & seed, 4);
-    sha.UpdateBigNumbers(&K, NULL);
+    sha.UpdateBigNumbers(&K, nullptr);
     sha.Finalize();
 
-    if (memcmp(sha.GetDigest(), digest, 20))
+    if (sha.GetDigest() != digest)
```

Comment:
This is not ok! Comparison of 2 arrays. At re-log, this comaprison will always trigger AUTH_FAILED player never reconnects

Reply:
right, for this std::array has to be involved... ech, bits of outdated content... will correct this. This comparition is valid between std::array.